app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -874,22 +878,17 @@ public boolean onOptionsItemSelected(MenuItem item) {
             invalidateOptionsMenu();
 
             return true;
-        } else if (id == R.id.action_archive) {
-            DBHelper.updateLoyaltyCardArchiveStatus(database, loyaltyCardId, 1);
-            Toast.makeText(LoyaltyCardViewActivity.this, R.string.archived, Toast.LENGTH_LONG).show();
-
-            ShortcutHelper.removeShortcut(LoyaltyCardViewActivity.this, loyaltyCardId);
-            new ListWidget().updateAll(LoyaltyCardViewActivity.this);
-
-            // Re-init loyaltyCard with new data from DB
-            onResume();
-            invalidateOptionsMenu();
-
-            return true;
-        } else if (id == R.id.action_unarchive) {
-            DBHelper.updateLoyaltyCardArchiveStatus(database, loyaltyCardId, 0);
-            Toast.makeText(LoyaltyCardViewActivity.this, R.string.unarchived, Toast.LENGTH_LONG).show();
+        } else if (id == R.id.action_archive_unarchive) {
+            if(loyaltyCard.archiveStatus == 0){
+                DBHelper.updateLoyaltyCardArchiveStatus(database, loyaltyCardId, 1);
+                Toast.makeText(LoyaltyCardViewActivity.this, R.string.archived, Toast.LENGTH_LONG).show();
 
+                ShortcutHelper.removeShortcut(LoyaltyCardViewActivity.this, loyaltyCardId);
+                new ListWidget().updateAll(LoyaltyCardViewActivity.this);
+            }else{
+                DBHelper.updateLoyaltyCardArchiveStatus(database, loyaltyCardId, 0);
+                Toast.makeText(LoyaltyCardViewActivity.this, R.string.unarchived, Toast.LENGTH_LONG).show();
+            }
```

Comment:
Inconsistent spacing with rest of the codebase

```suggestion
            if (loyaltyCard.archiveStatus == 0) {
                DBHelper.updateLoyaltyCardArchiveStatus(database, loyaltyCardId, 1);
                Toast.makeText(LoyaltyCardViewActivity.this, R.string.archived, Toast.LENGTH_LONG).show();

                ShortcutHelper.removeShortcut(LoyaltyCardViewActivity.this, loyaltyCardId);
                new ListWidget().updateAll(LoyaltyCardViewActivity.this);
            } else {
                DBHelper.updateLoyaltyCardArchiveStatus(database, loyaltyCardId, 0);
                Toast.makeText(LoyaltyCardViewActivity.this, R.string.unarchived, Toast.LENGTH_LONG).show();
            }
```